src/scss/parse.js

```diff
@@ -4038,106 +4023,257 @@ function _checkUnicodeWildcard(i) {
 
-    uri = uri.concat(getSC());
+    if (checkIdent(pos)) getIdent();
+    else if (checkNumber(pos)) getNumber();
+    else if (_checkUriRawCharacters(pos)) pos += 1;
+    // As comments are not allowed within the uri token, include as part of node
+    else if (l = checkCommentML(pos)) pos += l;
+    else break;
   }
 
-  token = tokens[startPos];
-  var line = token.ln;
-  var column = token.col;
-  var end = getLastPosition(uri, line, column, 1);
-  pos++;
+  content = joinValues(startPos, pos - 1);
 
-  return newNode(NodeType.UriType, uri, token.ln, token.col, end);
+  return newNode(type, content, line, column);
 }
 
 /**
- * @param {Number} i Token's index number
- * @returns {Number}
+ * Check for a raw URI
+ * (1) `url(http://foo/bar.png)`
+ * (2) `url(http://foo/#{bar}.png)`
+ * (3) `url(#{foo}/bar.png)`
+ * @param {number} i Token's index number
+ * @returns {number} Length of URI node
  */
-function checkUriContent(i) {
-  return checkUri1(i) ||
-      checkFunction(i);
+function checkUri1(i) {
+  const start = i;
+  let l;
+  let isRaw = false;
+
+  if (l = checkSC(i)) i += l;
+
+  while (i < tokensLength) {
+    if (checkVariable(i) || checkFunction(i)) return 0;
+
+    if (l = checkUriRaw(i)) {
+      isRaw = true;
+      i += l;
+    }
+    else if (l = checkInterpolation(i)) i += l;
+    else break;
+  }
+
+  if (l = checkSC(i)) i += l;
+
+  // Check if URI contains a raw node
+  if (!isRaw) return 0;
+
+  return i - start;
 }
 
 /**
- * @returns {Array}
+ * Get a raw URI node
+ * @returns {Node}
  */
-function getUriContent() {
-  if (checkUri1(pos)) return getString();
-  else if (checkFunction(pos)) return getFunction();
+function getUri1() {
+  const startPos = pos;
+  const type = NodeType.UriType;
+  const token = tokens[startPos];
+  const line = token.ln;
+  const column = token.col;
+  let content = [];
+  let end;
+
+  // Skip url and (
+  pos += 2;
+
+  if (checkSC(pos)) content = content.concat(getSC());
+
+  while (pos < tokensLength) {
+    if (checkUriRaw(pos)) content.push(getUriRaw());
+    else if (checkInterpolation(pos)) content.push(getInterpolation());
+    else break;
+  }
+
+  if (checkSC(pos)) content = content.concat(getSC());
+
+  end = getLastPosition(content, line, column, 1);
+
+  // Skip )
+  pos++;
+
+  return newNode(type, content, line, column, end);
 }
 
 /**
- * @param {Number} i Token's index number
- * @returns {Number}
+ * Check for a non-raw URI
+ * (1) `url('http://foo/bar.png')`
+ * (2) `url('http://foo/'#{bar}.png)`
+ * (3) `url(#{foo}'/bar.png')`
+ * @param {number} i Token's index number
+ * @returns {number} Length of URI node
  */
-function checkUri1(i) {
-  let start = i;
+function checkUri2(i) {
+  const start = i;
   let l;
 
-  if (i >= tokensLength) return 0;
+  while (i < tokensLength) {
+    if (l = checkSC(i)) i += l;
+    else if (l = checkString(i)) i += l;
+    else if (l = checkFunction(i)) i += l;
+    else if (l = checkUnary(i)) i += l;
```

Comment:
Unary operator is the one that gives sign to a number, for example, `-3` or `+8`. I think it can't be applied to uris but I guess there isn't anything more suitable in the code now :)